erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -2033,31 +2058,38 @@ func (hph *HexPatriciaHashed) Process(ctx context.Context, updates *Updates, log
 
 		default:
 		}
-
+		hph.metrics.Accounts.KeyRegister(plainKey)
 		if hph.trace {
 			fmt.Printf("\n%d/%d) plainKey [%x] hashedKey [%x] currentKey [%x]\n", ki+1, updatesCount, plainKey, hashedKey, hph.currentKey[:hph.currentKeyLen])
 		}
 		// Keep folding until the currentKey is the prefix of the key we modify
 		for hph.needFolding(hashedKey) {
+			startFold := hph.metrics.Now()
+			hph.metrics.Accounts.FoldsInc(hph.metrics.Accounts.currentPlainKey)
 			if err := hph.fold(); err != nil {
 				return fmt.Errorf("fold: %w", err)
 			}
+			hph.metrics.TotalFoldingTimeInc(startFold)
 		}
 		// Now unfold until we step on an empty cell
 		for unfolding := hph.needUnfolding(hashedKey); unfolding > 0; unfolding = hph.needUnfolding(hashedKey) {
+			unfoldDone := hph.metrics.StartUnfolding(hph.metrics.Accounts.currentPlainKey)
```

Comment:
here you used similar suggestion but for folding you saying its not related and would look ugly?